noir/compiler/noirc_evaluator/src/ssa/acir_gen/acir_ir/acir_variable.rs

```diff
@@ -1290,6 +1290,7 @@ impl AcirContext {
         inputs: Vec<Witness>,
         warnings: Vec<SsaReport>,
     ) -> GeneratedAcir {
+        self.acir_ir.decrement_witness_index();
```

Comment:
I think we should avoid this decrementing witness_index function so that we can uphold the invariant that the witness index is always increasing. I've not read into why this is needed yet, just a passing comment

Reply:
Yeah I will refactor how we operate with this `current_witness_index` as right now there is a bit of ambiguity.

Reply:
I switched to using `Option<u32>` for the `current_witness_index`, remove the `decrement_witness_index` method, and switched back to incrementing before returning the index in `next_witness_index`. We want the `current_witness_index` when we finish generating the ACIR to be the true final generated witness. With the way this was previously written the final index was the next available witness index, so I was decrementing as to not break the ACIR transformers.